openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/CollectionDAO.java

```diff
@@ -517,12 +567,12 @@ List<String> listBefore(
                 + ") er "
                 + "on ce.id = er.toId "
                 + "<sqlCondition> AND "
-                + "<nameColumn> > :after "
-                + "ORDER BY <nameColumn> "
+                + "JSON_EXTRACT(ce.json, concat('$.',:nameColumn)) > :after "
```

Comment:
Is there a way to avoid multiple JSON_EXTRACTs?

Reply:
fixed it